### ISSUE TITLE
creepjs doesn't trust me at all

### ISSUE THREAD

**Issue by mannok** [CheshireCaat/puppeteer-with-fingerprints#31]
I use the sample code directly to test with headful mode, code as below. However, creepjs assess me with 0% score... Is there anything I missed?

```
import { plugin } from "puppeteer-with-fingerprints";

(async () => {
  // Get a fingerprint from the server:
  const fingerprint = await plugin.fetch('', {
    tags: ['Microsoft Windows', 'Chrome'],
  });

  // Apply fingerprint:
  plugin.useFingerprint(fingerprint);

  // Launch the browser instance:
  const browser = await plugin.launch({ headless: false });

  // The rest of the code is the same as for a standard `puppeteer` library:
  const page = await browser.newPage();
  await page.goto('https://abrahamjuliot.github.io/creepjs');

  // Print the browser viewport size:
  console.log(
    'Viewport:',
    await page.evaluate(() => ({
      deviceScaleFactor: window.devicePixelRatio,
      width: document.documentElement.clientWidth,
      height: document.documentElement.clientHeight,
    }))
  );

  await browser.close();
})();
```

Please see screenshot below
![image](https://github.com/CheshireCaat/puppeteer-with-fingerprints/assets/458220/77634dee-765e-4252-8115-6977d226b5a6)

**Comment by bablosoft** [comment]
In order to bypass creepjs test, you need to use PerfectCanvas, it would be nice to install FontPack also.

https://wiki.bablosoft.com/doku.php?id=perfectcanvas

https://wiki.bablosoft.com/doku.php?id=fontpack

You also need to obtain fingerprint with same version as current Chrome version.

It works at least with BAS.


<img width="671" alt="NVIDIA_Share_2023-07-26_12-41-39" src="https://github.com/CheshireCaat/puppeteer-with-fingerprints/assets/19365536/0abdc56d-6e4b-4d41-a1cc-2df6cfc5ce30">
<img width="1044" alt="NVIDIA_Share_2023-07-26_12-45-35" src="https://github.com/CheshireCaat/puppeteer-with-fingerprints/assets/19365536/f2f03470-2da9-45ee-886a-6b7a01a845cc">

**Comment by bablosoft** [comment]
> I am interested in pro version, but I wish I could ensure it can really bypass creepjs from my side first.

I've tested it only with BAS (https://bablosoft.com/shop/BrowserAutomationStudio) It passes correctly. As you see on screenshot.

Note, that only using paid version is not enough, you need to use PerfectCanvas and FontPack.

> May I know if there is a trail license or refund policy?

There is only demo - using empty key.

Refund is available if key is not used, or if there is critical issue which is not fixed during 5 working days.

https://community.bablosoft.com/topic/12199 - but we have researching site protection research program and it is free.

> btw, what do you mean by current chrome version? You mean the latest version or the one installed in my PC?

puppeteer-with-fingerprints ships with several browser versions, right now it is 115 and 114. You can select browser version like this:

https://github.com/CheshireCaat/puppeteer-with-fingerprints#configuring-browser-version

Selected browser version must match browser version from fingerprint, otherwise discrepancies could be detected.

I think, I'll pin this ticket.

**Comment by mannok** [comment]
Thanks @bablosoft , I wonder if I required to use perfectcanvas and fontpack with BAS? Can I use them without BAS?   Since I found all the tutortial you provided is for integration with BAS

**Comment by bablosoft** [comment]
> Can I use them without BAS?

Yes, you need to specify PerfectCanvas when fetching fingerprint.

https://github.com/CheshireCaat/browser-with-fingerprints/blob/master/src/index.d.ts#L189

FontPack and PerfectCanvas are enabled by default:

https://github.com/CheshireCaat/browser-with-fingerprints/blob/2b6c5dd274b84ee74aa3414e2cde1238b0d82dfe/src/index.d.ts#L63

https://github.com/CheshireCaat/browser-with-fingerprints/blob/2b6c5dd274b84ee74aa3414e2cde1238b0d82dfe/src/index.d.ts#L72

Also note, BAS contains more features regarding anonymity than puppeteer-with-fingerprints, for example it protects from detection of javascript execution. So if you have really strong security, you should consider using it.

--------------------------------

> How can we assume the PerfectCanvas request is required only once per site? What if the site ask us to draw another image/text or ask for another quest next time?

In that case it would be much harder for site to compute uniqueness of value, each new challenge means collecting new database. If request is random each time, you are safe to use noise or real values for that particular item.

CanvasInspector also has feature to show if request is changed. You can try by yourself, it is free.

<img width="1423" alt="NVIDIA_Share_2023-07-26_14-55-00" src="https://github.com/CheshireCaat/puppeteer-with-fingerprints/assets/19365536/0e26786a-c810-473a-b24d-a9bd4998079f">

<img width="1423" alt="NVIDIA_Share_2023-07-26_14-55-49" src="https://github.com/CheshireCaat/puppeteer-with-fingerprints/assets/19365536/26be3115-fb53-400a-b8c6-76dec8daedae">

**Comment by mannok** [comment]
@bablosoft wow!   So detail and thanks for your clear explanation. Let me try in depth with BAS and all component from bablosoft.  If everything goes fine  then my team will adopt this for sure.    Thanks a lot!
